src/unused_finder/node_visitor.rs

```diff
@@ -18,21 +18,45 @@ use swc_core::{
         visit::{Visit, VisitWith},
     },
 };
+
+#[derive(Debug, Default, Eq, PartialEq, Clone, Hash)]
+pub struct ExportedItemMetadata {
+    pub export_type: ExportType,
+    pub span: Span,
+    pub allow_unused: bool,
+}
+
+impl ExportedItemMetadata {
+    pub fn new(export_type: ExportType, span: Span, allow_unused: bool) -> Self {
+        Self {
+            export_type,
+            span,
+            allow_unused,
+        }
+    }
+}
+
 #[derive(Debug, PartialEq, Eq, Hash, Clone)]
-pub enum ExportedItem {
+pub enum ExportType {
```

Comment:
soft suggestion, consider `ExportKind` here insteaf of `ExportType`, for possible confusion with the `export type` syntax and typeonly-exports from typescript files.